src/app/state/multiaccounts/index.ts

```diff
@@ -1,39 +1,49 @@
 import { PayloadAction } from '@reduxjs/toolkit'
 import { ErrorPayload } from 'types/errors'
 import { createSlice } from 'utils/@reduxjs/toolkit'
-import { LedgerAccount, LedgerState, LedgerStep } from './types'
+import { MultiAccountsListAccount, MultiAccountsState, MultiAccountsStep } from './types'
 
-export const initialState: LedgerState = { accounts: [] }
+export const initialState: MultiAccountsState = { accounts: [] }
 
 const slice = createSlice({
-  name: 'ledger',
+  name: 'multiAccounts',
   initialState,
   reducers: {
     clear(state, action: PayloadAction<void>) {
       state.accounts = []
       state.error = undefined
+      state.mnemonic = undefined
+      state.step = undefined
     },
-    enumerateAccounts(state, action: PayloadAction<void>) {
+    enumerateAccountsFromLedger(state, action: PayloadAction<void>) {
+      state.step = undefined
+      state.accounts = []
+    },
+    enumerateAccountsFromMnemonic(state, action: PayloadAction<string>) {
       state.step = undefined
       state.accounts = []
     },
     toggleAccount(state, action: PayloadAction<number>) {
       const index = action.payload
       state.accounts[index].selected = !state.accounts[index].selected
     },
-    accountsListed(state, action: PayloadAction<LedgerAccount[]>) {
+    accountsListed(state, action: PayloadAction<MultiAccountsListAccount[]>) {
       state.accounts = action.payload
     },
-    setStep(state, action: PayloadAction<LedgerStep>) {
+    setMnemonic(state, action: PayloadAction<string>) {
+      state.mnemonic = action.payload
+    },
```

Comment:
I don't like this. I think better solution would be for MultiAccountsSelector to take something like AddWalletPayload[], so there isn't a need to call `getAccountSigner(mnemonic, index)` twice (when enumerating and when opening) and risk a discrepancy in indexes

Reply:
To render a selection modal we don't need public and private keys. We may not even need those keys as user can cancel an action/close modal. Question is do we want to generate private keys upfront or later when they actually needed.